### ISSUE TITLE
Build fails:  too many filenames given

### ISSUE THREAD

**Issue by mkrusemark** [emqx/quic#46]
Build fails:

#12 182.0 [ 82%] Building C object CMakeFiles/quicer_static.dir/c_src/quicer_ctx.c.o
#12 182.0 cc1: error: too many filenames given; type 'cc1 --help' for usage
#12 182.0 cc1: fatal error: CMakeFiles/quicer_static.dir/c_src/quicer_nif.c.d: No such file or directory
#12 182.0 compilation terminated.

GCC-11 but even GCC-10 fails
CMake 3.20

**Comment by qzhuyan** [comment]
Can you provide full outputs?

**Comment by brianmay** [comment]
I am seeing this error. But only when trying to build using github actions.

https://github.com/brianmay/exmqtt/runs/3064947467

I am finding it somewhat weird in that often some files will compile fine, then suddenly it stops working for no reason.

In debugging this, I noticed this error:

```
sh: 1: nprocs: not found
```

I believe the correct command in `nproc` not `nprocs`. I tried to search for `nprocs` but found nothing. Unfortunately this is not the cause of the cc1 error. I created a symlink from nprocs to nproc, and it fixes the nprocs error, but it still fails to build.

**Comment by brianmay** [comment]
For comparison, builds on my desktop running Debian/bullseye work fine.

**Comment by brianmay** [comment]
Here is another example with quic version 0.0.6 (I think the previous example used 0.0.3).

https://github.com/brianmay/exmqtt/runs/3070131702?check_suite_focus=true

**Comment by qzhuyan** [comment]
> Here is another example with quic version 0.0.6 (I think the previous example used 0.0.3).
> 
> https://github.com/brianmay/exmqtt/runs/3070131702

@brianmay thanks for your log, I guess we can narrow down the issue to the version of CMAKE.

according to https://github.com/actions/virtual-environments/blame/main/images/linux/Ubuntu2004-README.md#L62
the CMAKE version in use is CMake 3.20.5 

 Could you try to downgrade CMAKE version to 3.16 as in this link?

https://github.com/qzhuyan/quic/blob/main/.github/workflows/main.yml#L47-L50

I will see if I can reproduce the issue with CMake 3.20.5

**Comment by brianmay** [comment]
@qzhuyan I was thinking cmake before, but got confused and though cmake wasn't used :-(

Downgrading cmake does seem to help.

Also note that cmake in Debian/bullseye - which is working is 3.18.4. So guessing something after that version breaks.

**Comment by qzhuyan** [comment]
@brianmay thanks for the update

I managed to reproduce the issue with CMake 3.20

> Also note that CMake in Debian/bullseye - which is working is 3.18.4. So guessing something after that version breaks.

We have tested with CMake up till 3.19.2 in other CI envs and they work well.

**Comment by qzhuyan** [comment]
note-taking:
build with 

```
#CMake 3.20.5
make VERBOSE=1 
```

Get something about

```
...
/usr/bin/cc -DCX_PLATFORM_LINUX -DQUIC_DISABLE_CLIENT_CERT_TESTS -DQUIC_DISABLE_DEFERRED_CERT_TESTS -DQUIC_EVENTS_STUB -DQUIC_LOGS_STUB -DVER_BUILD_ID=0 -DVER_SUFFIX=-private -D_GNU_SOURCE -I/usr/local/lib/erlang/usr/include -I/quic/submodules/msquic/src/inc -isystem /quic/c_build/submodules/msquic/openssl/include -g -DSO_ATTACH_REUSEPORT_CBPF=51 -og -ggdb3 -Wno-unused-variable -fms-extensions -fPIC -pthread -Werror -Wall -Wextra -Wformat=2 -Wno-type-limits -Wno-unknown-pragmas -Wno-multichar -Wno-missing-field-initializers -MD -MT CMakeFiles/quicer_static.dir/c_src/quicer_connection.c.o -MF CMakeFiles/quicer_static.dir/c_src/quicer_connection.c.o.d -o CMakeFiles/quicer_static.dir/c_src/quicer_connection.c.o -c /quic/c_src/quicer_connection.c
/usr/bin/cc -DCX_PLATFORM_LINUX -DQUIC_DISABLE_CLIENT_CERT_TESTS -DQUIC_DISABLE_DEFERRED_CERT_TESTS -DQUIC_EVENTS_STUB -DQUIC_LOGS_STUB -DVER_BUILD_ID=0 -DVER_SUFFIX=-private -D_GNU_SOURCE -I/usr/local/lib/erlang/usr/include -I/quic/submodules/msquic/src/inc -isystem /quic/c_build/submodules/msquic/openssl/include -g -DSO_ATTACH_REUSEPORT_CBPF=51 -og -ggdb3 -Wno-unused-variable -fms-extensions -fPIC -pthread -Werror -Wall -Wextra -Wformat=2 -Wno-type-limits -Wno-unknown-pragmas -Wno-multichar -Wno-missing-field-initializers -MD -MT CMakeFiles/quicer_static.dir/c_src/quicer_stream.c.o -MF CMakeFiles/quicer_static.dir/c_src/quicer_stream.c.o.d -o CMakeFiles/quicer_static.dir/c_src/quicer_stream.c.o -c /quic/c_src/quicer_stream.c
cc1: error: too many filenames given.  Type cc1 --help for usage
...
```

**Comment by qzhuyan** [comment]
released in 0.0.7
